packages/cells/src/cells/date-picker-cell.tsx

```diff
@@ -1,15 +1,80 @@
-import * as React from "react";
-import { CustomCell, CustomRenderer, drawTextCell, GridCellKind } from "@glideapps/glide-data-grid";
+import React from "react";
+import { assertNever } from "../../../core/src/common/support";
+import {
+    CustomCell,
+    CustomRenderer,
+    drawTextCell,
+    GridCellKind,
+    ProvideEditorCallback,
+} from "@glideapps/glide-data-grid";
 
 interface DatePickerCellProps {
     readonly kind: "date-picker-cell";
     readonly date: Date | undefined;
     readonly displayDate: string;
-    readonly format: "date" | "datetime-local";
+    readonly format: DateType;
 }
 
+export type DateType = "date" | "time" | "datetime-local";
+
+export const formatValueForHTMLInput = (format: DateType, date: Date): string => {
+    switch (format) {
+        case "date":
+            return date.toISOString().split("T")[0];
+        case "datetime-local":
+            return date.toISOString().replace("Z", "");
+        case "time":
+            return date.toISOString().split("T")[1].replace("Z", "");
+        default:
+            assertNever(format);
+    }
+};
+
 export type DatePickerCell = CustomCell<DatePickerCellProps>;
 
+const Editor: ReturnType<ProvideEditorCallback<DatePickerCell>> = cell => {
+    const cellData = cell.value.data;
+    const { date, displayDate, format } = cellData;
+    const cellDataDate =
+        cellData.date ??
+        // now as date with adjusted timezone to allow for displayDate to be correct
+        new Date(new Date().getTime() - new Date().getTimezoneOffset() * 60000);
+    const value = formatValueForHTMLInput(format, cellDataDate);
+
+    return (
+        <input
+            required
+            style={{ minHeight: 26, border: "none", outline: "none" }}
+            type={format}
+            value={value}
+            autoFocus={true}
+            onChange={event => {
+                // handle when clear is clicked and value has been wiped
+                if (event.target.value === "") {
+                    cell.onChange({
+                        ...cell.value,
+                        data: {
+                            ...cell.value.data,
+                            // attempt to reset to cached date
+                            date: date !== undefined ? date : new Date(displayDate),
```

Comment:
One question here would be if we even want to have this behavior. I think we have the following options in case the user selects clear in the date picker dialog:
1) reset the value (`date`) to the current value of the cell (this is what the implementation from above is doing)
2) set the value (`date`) to `undefined` and thereby clearing the cell. 

I think both options seem to be reasonable, but from playing around a bit with the input components, I have a slight preference for `2` (clearing the cell value).

Reply:
This will unfortunately set the underlying date to undefined and make the cell null. If the cell has a value and then you do a clear, I would expect as a user that it just clears the calendar and not the actual cell, especially since it doesn't close the calendar / editor.

Reply:
Thanks Lukas :)